app/migrate_module_account.go

```diff
@@ -0,0 +1,78 @@
+package app
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authkeeper "github.com/cosmos/cosmos-sdk/x/auth/keeper"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+)
+
+// MigratePreInitializedModuleAccounts migrates module accounts that were pre-initialized as BaseAccounts to ModuleAccounts,
+// or creates new module accounts if they don't exist.
+func MigratePreInitializedModuleAccounts(
+	ctx sdk.Context,
+	ak authkeeper.AccountKeeper,
+	moduleAccountsToInitialize []string,
+) error {
+	for _, module := range moduleAccountsToInitialize {
+		addr, perms := ak.GetModuleAddressAndPermissions(module)
+		if addr == nil {
+			return fmt.Errorf(
+				"failed to get module address and permissions for module %s",
+				module,
+			)
+		}
+
+		acc := ak.GetAccount(ctx, addr)
+		// The account has not been initialized yet
+		if acc == nil {
+			initModuleAccount(ctx, ak, module, perms...)
+			continue
+		}
```

Comment:
is this going to interfere with the regular instantiation of modules?

Reply:
nope, it will be a no-op for regular instantiation if account already exists